src/components/navigation/document-view.tsx

```diff
@@ -111,54 +131,121 @@ export const DocumentView = observer(function DocumentView({tabSpec, subTab}: IP
+    }
+    if (tabSpec.tab === "my-work") {
+      return (primaryDocIndex === 0
+                || (secondaryDocIndex === 0 && primaryDocIndex === numStarredDocs - 1));
     }
   };
 
-  const hideLeftFlipper = () => {
-    if (tabSpec.tab === "class-work") return (currentOpenDocIndex === numStarredDocs - 1);
-    if (tabSpec.tab === "my-work") return (currentOpenDocIndex === 0);
+  const hideRightFlipper = (position?: string) => {
+    const primaryDocIndex = position === "secondary" ? currentOpenSecondaryDocIndex : currentOpenDocIndex;
+    const secondaryDocIndex = position === "secondary" ? currentOpenDocIndex : currentOpenSecondaryDocIndex;
+    if (tabSpec.tab === "class-work") {
+      return (primaryDocIndex === 0 || (secondaryDocIndex === 0 && primaryDocIndex === 1));
+    }
+    if (tabSpec.tab === "my-work") {
+      return (primaryDocIndex === 1 || (secondaryDocIndex === 1 && primaryDocIndex === 2));
+    }
   };
-  const hideRightFlipper = () => {
-    if (tabSpec.tab === "class-work") return (currentOpenDocIndex === 0);
-    if (tabSpec.tab === "my-work") return (currentOpenDocIndex === 1);
+
+  const getDisplayTitle = (document: DocumentModelType) => {
+    const documentOwner = classStore.users.get(document.uid);
+    const documentTitle = getDocumentDisplayTitle(document, appConfigStore, problemStore);
+    return openSecondaryDocument && documentOwner ? documentOwner.fullName + " " + documentTitle : documentTitle;
   };
 
   return (
     <div className="scroller-and-document">
       { isStarredTab &&
         <DocumentBrowserScroller subTab={subTab} tabSpec={tabSpec} openDocumentKey={openDocumentKey}
-            onSelectDocument={handleSelectDocument} />
+            openSecondaryDocumentKey={openSecondaryDocumentKey} onSelectDocument={handleSelectDocument} />
       }
       {skipDocument
         ? null
         : <div className="document-area">
-            <div className={`document-header ${tabSpec.tab} ${sectionClass}`} onClick={() => ui.setSelectedTile()}>
-              <div className={`document-title`}>
-                {getDocumentDisplayTitle(openDocument, appConfigStore, problemStore)}
+            <div className={`focus-document ${openSecondaryDocument ? "primary" : ""}`}>
+              <div className={`document-header ${tabSpec.tab} ${sectionClass}
+                              ${openSecondaryDocument ? "primary" : ""}`} onClick={() => ui.setSelectedTile()}>
+                <div className={`document-title`}>
+                  {getDisplayTitle(openDocument)}
+                </div>
+                {(!openDocument.isRemote)
+                    && editButton(tabSpec.tab, sectionClass, openDocument)}
+              </div>
+              <div className="scroll-arrow-button-wrapper left">
+                <ScrollButton side={"left"} tab={tabSpec.tab} onScroll={() => handleChangeDocument(1)}
+                    hidden={!isStarredTab || (isStarredTab && hideLeftFlipper())} />
+              </div>
```

Comment:
This code is now duplicated 4 times. You should make it a named component like you did for `ScrollEndControl`.